watertap/costing/unit_models/ion_exchange.py

```diff
@@ -268,6 +270,10 @@ def cost_ion_exchange(blk):
 
     elif ix_type == "anion":
         resin_cost = ion_exchange_params.anion_exchange_resin_cost
+    else:
+        raise WaterTapDeveloperError(
```

Comment:
This gets checked in the unit model directly via checking of the charge on a component. It should be essentially impossible to get to this point.

... but maybe that is the point of a developer error?